### PR TITLE
Support WIF in examples workflows

### PR DESCRIPTION
**Pull request by FrodoTheTrue**

### CODE REVIEW

**Review by sethvargo — APPROVED** [review]

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml [hunk 1]
```diff
@@ -32,13 +32,24 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@master
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
```
Comment (bharathkkb) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:38]:
```suggestion
      uses: 'google-github-actions/auth@v0.4.1'
```

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml [hunk 2]
```diff
@@ -32,13 +32,24 @@ jobs:
...
       with:
-        version: '286.0.0'
-        service_account_email: ${{ secrets.RUN_SA_EMAIL }}
-        service_account_key: ${{ secrets.RUN_SA_KEY }}
-        project_id: ${{ secrets.RUN_PROJECT }}
+        token_format: 'access_token'
```
Comment (bharathkkb) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:40]: I don't think we need to create an access token, setup-gcloud picks up just the WIF creds file.

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml (hunk 1 above)
Comment (sethvargo) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:38]: Can you pin to `@v0` instead?

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml [hunk 3]
```diff
@@ -32,13 +32,24 @@ jobs:
...
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
+    #   uses: google-github-actions/auth@v0.4.0
```
Comment (sethvargo) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:46]: `@v0`

File: example-workflows/cloud-run/cloud-run.yml [hunk 4]
```diff
@@ -30,11 +30,24 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    - name: Setup Cloud SDK
-      uses: google-github-actions/setup-gcloud@v0.2.0
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
```
Comment (sethvargo) [inline example-workflows/cloud-run/cloud-run.yml:36]: `@v0`

File: example-workflows/cloud-run/cloud-run.yml [hunk 5]
```diff
@@ -30,11 +30,24 @@ jobs:
...
       with:
-        project_id: ${{ env.PROJECT_ID }}
-        service_account_key: ${{ secrets.GCP_SA_KEY }}
+        token_format: 'access_token'
```
Comment (sethvargo) [inline example-workflows/cloud-run/cloud-run.yml:38]: Not needed

File: example-workflows/gae/app-engine.yml [hunk 6]
```diff
@@ -27,11 +27,24 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Setup Cloud SDK
-        uses: google-github-actions/setup-gcloud@v0.2.0
+      # Configure Workload Identity Federation and generate an access token.
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0.4.0'
```
Comment (sethvargo) [inline example-workflows/gae/app-engine.yml:33]: `@v0`

File: example-workflows/gae/app-engine.yml [hunk 7]
```diff
@@ -27,11 +27,24 @@ jobs:
...
         with:
-          project_id: ${{ env.PROJECT_ID }}
-          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          token_format: 'access_token'
```
Comment (sethvargo) [inline example-workflows/gae/app-engine.yml:35]: Not needed

File: example-workflows/gce/.github/workflows/gce.yaml [hunk 8]
```diff
@@ -33,12 +33,24 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@master
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
```
Comment (sethvargo) [inline example-workflows/gce/.github/workflows/gce.yaml:39]: `@v0`

File: example-workflows/gce/.github/workflows/gce.yaml [hunk 9]
```diff
@@ -33,12 +33,24 @@ jobs:
...
       with:
-        version: '290.0.1'
-        service_account_key: ${{ secrets.GCE_SA_KEY }}
-        project_id: ${{ secrets.GCE_PROJECT }}
+        token_format: 'access_token'
```
Comment (sethvargo) [inline example-workflows/gce/.github/workflows/gce.yaml:41]: Not needed

File: example-workflows/gce/.github/workflows/gce.yaml [hunk 10]
```diff
@@ -33,12 +33,24 @@ jobs:
...
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
+    #   uses: google-github-actions/auth@v0.4.0
```
Comment (sethvargo) [inline example-workflows/gce/.github/workflows/gce.yaml:47]: `@v0`

File: example-workflows/gae/app-engine.yml [hunk 11]
```diff
@@ -27,11 +27,24 @@ jobs:
...
+          workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+          service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+      # Alternative option - authentication via credentials json
+      # - id: auth
+      #   uses: google-github-actions/auth@v0.4.0
```
Comment (sethvargo) [inline example-workflows/gae/app-engine.yml:41]: `@v0`

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 12]
```diff
@@ -28,6 +28,10 @@ env:
   REPOSITORY: samples # TODO: update to Artifact Registry docker repository
   IMAGE: static-site
 
+permissions:
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:31]: It'd be better to put this inside the specific job that needs it, that way it's not exposed in all jobs.

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 13]
```diff
@@ -37,11 +41,24 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@master
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:47]: `@v0`

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 14]
```diff
@@ -37,11 +41,24 @@ jobs:
...
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        token_format: 'access_token'
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
+    #   uses: google-github-actions/auth@v0.4.0
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:55]: `@v0`

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 15]
```diff
@@ -37,11 +41,24 @@ jobs:
...
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@master
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        token_format: 'access_token'
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:49]: Not needed

File: example-workflows/gke/.github/workflows/gke.yml [hunk 16]
```diff
@@ -34,11 +34,24 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@master
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
```
Comment (sethvargo) [inline example-workflows/gke/.github/workflows/gke.yml:40]: `@v0`

File: example-workflows/gke/.github/workflows/gke.yml [hunk 17]
```diff
@@ -34,11 +34,24 @@ jobs:
...
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        token_format: 'access_token'
```
Comment (sethvargo) [inline example-workflows/gke/.github/workflows/gke.yml:42]: Not needed

File: example-workflows/gke/.github/workflows/gke.yml [hunk 18]
```diff
@@ -34,11 +34,24 @@ jobs:
...
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
+    #   uses: google-github-actions/auth@v0.4.0
```
Comment (sethvargo) [inline example-workflows/gke/.github/workflows/gke.yml:48]: `@v0`

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml [hunk 19]
```diff
@@ -28,17 +28,32 @@ jobs:
...
       with:
-        version: '286.0.0'
-        service_account_email: ${{ secrets.RUN_SA_EMAIL }}
-        service_account_key: ${{ secrets.RUN_SA_KEY }}
-        project_id: ${{ secrets.RUN_PROJECT }}
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
```
Comment (sethvargo) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:49]: Please quote these (even though they are commented)

File: example-workflows/cloud-build/.github/workflows/cloud-build.yml [hunk 20]
```diff
@@ -28,17 +28,32 @@ jobs:
...
+    #   uses: google-github-actions/auth@v0
+    #   with:
+    #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+    # Setup gcloud CLI
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/cloud-build/.github/workflows/cloud-build.yml:56]:
```suggestion
      uses: google-github-actions/setup-gcloud@v0
```

File: example-workflows/cloud-run/cloud-run.yml [hunk 21]
```diff
@@ -26,15 +26,33 @@ env:
...
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@@v0'
       with:
-        project_id: ${{ env.PROJECT_ID }}
-        service_account_key: ${{ secrets.GCP_SA_KEY }}
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
```
Comment (sethvargo) [inline example-workflows/cloud-run/cloud-run.yml:47]: Please quote these (even though they are commented)

File: example-workflows/cloud-run/cloud-run.yml [hunk 22]
```diff
@@ -26,15 +26,33 @@ env:
...
     steps:
     - name: Checkout
       uses: actions/checkout@v2
 
-    - name: Setup Cloud SDK
-      uses: google-github-actions/setup-gcloud@v0.2.0
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@@v0'
```
Comment (sethvargo) [inline example-workflows/cloud-run/cloud-run.yml]:
```suggestion
      uses: 'google-github-actions/auth@v0'
```

File: example-workflows/cloud-run/cloud-run.yml [hunk 23]
```diff
@@ -26,15 +26,33 @@ env:
...
+
+    # Alternative option - authentication via credentials json
+    # - id: auth
+    #   uses: google-github-actions/auth@v0
+    #   with:
+    #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+    # Setup gcloud CLI
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/cloud-run/cloud-run.yml:55]:
```suggestion
      uses: google-github-actions/setup-gcloud@v0
```

File: example-workflows/gae/app-engine.yml [hunk 24]
```diff
@@ -23,15 +23,33 @@ jobs:
...
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0'
         with:
-          project_id: ${{ env.PROJECT_ID }}
-          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+          service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+      # Alternative option - authentication via credentials json
```
Comment (sethvargo) [inline example-workflows/gae/app-engine.yml:44]: Please quote these (even though they are commented)

File: example-workflows/gae/app-engine.yml [hunk 25]
```diff
@@ -23,15 +23,33 @@ jobs:
...
+      # - id: auth
+      #   uses: google-github-actions/auth@v0
+      #   with:
+      #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+      # Setup gcloud CLI
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/gae/app-engine.yml:52]:
```suggestion
        uses: google-github-actions/setup-gcloud@v0
```

File: example-workflows/gce/.github/workflows/gce.yaml [hunk 26]
```diff
@@ -29,16 +29,32 @@ jobs:
...
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0'
       with:
-        version: '290.0.1'
-        service_account_key: ${{ secrets.GCE_SA_KEY }}
-        project_id: ${{ secrets.GCE_PROJECT }}
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
```
Comment (sethvargo) [inline example-workflows/gce/.github/workflows/gce.yaml:49]: Please quote these (even though they are commented)

File: example-workflows/gce/.github/workflows/gce.yaml [hunk 27]
```diff
@@ -29,16 +29,32 @@ jobs:
...
+    # - id: auth
+    #   uses: google-github-actions/auth@v0
+    #   with:
+    #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+    # Setup gcloud CLI
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/gce/.github/workflows/gce.yaml:57]:
```suggestion
      uses: google-github-actions/setup-gcloud@v0
```

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 28]
```diff
@@ -33,15 +33,32 @@ jobs:
...
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0'
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:53]: Please quote these (even though they are commented)

File: example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml [hunk 29]
```diff
@@ -33,15 +33,32 @@ jobs:
...
+    # - id: auth
+    #   uses: google-github-actions/auth@v0
+    #   with:
+    #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+    # Setup gcloud CLI
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/gke-kustomize/.github/workflows/gke-kustomize.yml:61]:
```suggestion
      uses: google-github-actions/setup-gcloud@v0
```

File: example-workflows/gke/.github/workflows/gke.yml [hunk 30]
```diff
@@ -30,15 +30,32 @@ jobs:
...
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0'
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Alternative option - authentication via credentials json
```
Comment (sethvargo) [inline example-workflows/gke/.github/workflows/gke.yml:50]: Please quote these (even though they are commented)

File: example-workflows/gke/.github/workflows/gke.yml [hunk 31]
```diff
@@ -30,15 +30,32 @@ jobs:
...
+    # - id: auth
+    #   uses: google-github-actions/auth@v0
+    #   with:
+    #     credentials_json: ${{ secrets.GCP_CREDENTIALS }}
+
+    # Setup gcloud CLI
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v0.3.0
```
Comment (sethvargo) [inline example-workflows/gke/.github/workflows/gke.yml:58]:
```suggestion
      uses: google-github-actions/setup-gcloud@v0
```

**Review by sethvargo — CHANGES_REQUESTED** [review]

**Review by sethvargo — APPROVED** [review]

**Comment by FrodoTheTrue** [comment]
Connected PR to starter-workflows: https://github.com/actions/starter-workflows/pull/1327/files

**Pull request merged**